packages/node_modules/@ciscospark/widget-recents/README.md

```diff
@@ -0,0 +1,274 @@
+var widgetObject = ciscospark.widget(widgetEl);
+```
+
+##### Widget Teardown
+
+When a widget needs to be removed from the page you will want to call the `.remove()` method. This will close any network connections active and remove the widget from the DOM. You can also pass a callback as a parameter to the `.remove()` method. The method also returns a Promise that is thenable.
+
+The returned value, `removed`, is `true` if a matching widget has been removed, and is `false` no widget was found.
+
+``` js
+// Basic remove
+ciscospark.widget(widgetEl).remove();
+
+// With callback
+ciscospark.widget(widgetEl).remove(function(removed) {
+  console.log('removed!');
+});
+
+// With Promise
+ciscospark.widget(widgetEl).remove().then(function(removed) {
+  console.log('removed!');
+});
+```
+
+
+##### NOTE
+> If you are also using the Spark JS SDK on the same page, please be sure to load that before you load the widget scripts.
+
+
+### JSX
+
+Because our widgets are built using React, you'll be able to directly import the modules and components into your React app.
+
+``` js
+import RecentsWidget from '@ciscospark/widget-recents';
+
+ReactDOM.render(
+  <RecentsWidget
+    accessToken="YOUR_ACCESS_TOKEN"
+  />,
+  document.getElementById('ELEMENT')
+);
+```
+
+### Events
+
+The Recents widget exposes a few events for hooking into widget functionality.
+You can directly add DOM event listener like this:
+
+``` html
+<div
+  class="ciscospark-widget"
+  data-toggle="ciscospark-recents"
+  data-access-token="YOUR_ACCESS_TOKEN"
+  />
+<script>
+  document.getElementById('ciscospark-widget').addEventListener('EVENT_NAME', function(event) {
+    // Handle the event here
+    console.log(event.detail);
+  });
+</script>
+```
+
+If you are using *browser globals*, you can provide a callback parameter that will fire whenever any event occurs. You can filter the actions using the name provided like this:
+
+``` js
+var widgetEl = document.getElementById('my-ciscospark-widget');
+// Init a new widget
+ciscospark.widget(widgetEl).recentsWidget({
+  accessToken: 'YOUR_ACCESS_TOKEN',
+});
+
+function callback(name, detail) {
+  if (name === 'messages:created') {
+    // Perform an action if a new message has been created
+  }
+}
+```
+
+Or you can use the (`ampersand-events`)[https://github.com/AmpersandJS/ampersand-events] API to listen to events like this:
+
+``` js
+var widgetEl = document.getElementById('ciscospark-widget');
+ciscospark.widget(widgetEl).on('messages:created', function(e) {
+  console.log(e.detail);
+});
+```
+
+Currently available events:
+
+* `messages:created` - A new message has been posted
+* `messages:unread` - The conversation has unread messages
+* `messages:read` - The conversation's messages have been read
+* `calls:ringing` - An incoming call is ringing
+* `calls:connected` - A call has been answered is now connected
+* `calls:disconnected` - A active call has been disconnected
+
```

Comment:
Are we going to have any space based events? Maybe: 
* `space:added` - You were added to a new space
* `space:mentioned` - You were mentioned in a space